src/main/kotlin/org/opensearch/replication/action/stop/TransportInternalStopIndexReplicationAction.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.replication.action.stop
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.apache.logging.log4j.LogManager
+import org.opensearch.action.support.HandledTransportAction
+import org.opensearch.action.ActionRequest
+import org.opensearch.action.support.ActionFilters
+import org.opensearch.action.support.clustermanager.AcknowledgedResponse
+import org.opensearch.transport.client.Client
+import org.opensearch.common.inject.Inject
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver
+import org.opensearch.cluster.service.ClusterService
+import org.opensearch.commons.replication.action.ReplicationActions.STOP_REPLICATION_ACTION_TYPE
+import org.opensearch.commons.replication.action.StopIndexReplicationRequest
+import org.opensearch.commons.replication.action.ReplicationActions.INTERNAL_STOP_REPLICATION_ACTION_NAME
+import org.opensearch.commons.utils.recreateObject
+import org.opensearch.core.action.ActionListener
+import org.opensearch.replication.metadata.ReplicationMetadataManager
+import org.opensearch.replication.util.coroutineContext
+import org.opensearch.replication.util.stackTraceToString
+import org.opensearch.replication.util.suspendExecute
+import org.opensearch.tasks.Task
+import org.opensearch.threadpool.ThreadPool
+import org.opensearch.transport.TransportService
+
+/* Internal transport action used by Index Management plugin to invoke stop replication
+   It transforms the request, and invokes the stop replication action (TransportStopIndexReplicationAction) on it.
+ */
+class TransportInternalStopIndexReplicationAction @Inject constructor (

```

Comment:
Just curious, why did we require a separate internal action? What was the challenge with directly calling the existing one?